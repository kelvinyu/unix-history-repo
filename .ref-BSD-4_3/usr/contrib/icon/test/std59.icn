record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   write("any('aeiou',&lcase) ----> ",image(any('aeiou',&lcase)) | "none")
   write("any('aeiou',&ucase) ----> ",image(any('aeiou',&ucase)) | "none")
   write("every write(any('aeiou',&lcase,1 to 5,10 to 20)) ----> ",image(every write(any('aeiou',&lcase,1 to 5,10 to 20))) | "none")
   write("match(\"abc\",\"abcabcabcabc\") ----> ",image(match("abc","abcabcabcabc")) | "none")
   write("match(\"abc\",\"xabcabcabcabc\") ----> ",image(match("abc","xabcabcabcabc")) | "none")
   write("every write(match(\"abc\",\"xabcabcabcabc\",1 to 10,1 to 10)) ----> ",image(every write(match("abc","xabcabcabcabc",1 to 10,1 to 10))) | "none")
   write("upto('56d&',&lcase) ----> ",image(upto('56d&',&lcase)) | "none")
   write("upto('56d&',&ucase) ----> ",image(upto('56d&',&ucase)) | "none")
   write("upto('56d&',&lcase,15) ----> ",image(upto('56d&',&lcase,15)) | "none")
   write("many(&lcase,\"this is a Test\") ----> ",image(many(&lcase,"this is a Test")) | "none")
   write("many(&lcase,\"this is a Test\",5) ----> ",image(many(&lcase,"this is a Test",5)) | "none")
   write("many(&lcase,\"this is a Test\",5,9) ----> ",image(many(&lcase,"this is a Test",5,9)) | "none")
   write("find(\"aa\",\"xxaaaaaa\") ----> ",image(find("aa","xxaaaaaa")) | "none")
   write("every write(find(\"aa\",\"xxaaaaaa\")) ----> ",image(every write(find("aa","xxaaaaaa"))) | "none")
   write("every write(find(\"aa\",\"xxaaaaaa\",4,7)) ----> ",image(every write(find("aa","xxaaaaaa",4,7))) | "none")
   write("bal('-','(',')',\"-35\") ----> ",image(bal('-','(',')',"-35")) | "none")
   write("bal('+','(',')',\"((2*x)+3)+(5*y)\") ----> ",image(bal('+','(',')',"((2*x)+3)+(5*y)")) | "none")
   write("every write(bal('+','(',')',\"((2*x)+3)+(5*y)\",1 to 10)) ----> ",image(every write(bal('+','(',')',"((2*x)+3)+(5*y)",1 to 10))) | "none")
   write("bal('+','[','[',\"[[2*x[+3[+[5*y[\") ----> ",image(bal('+','[','[',"[[2*x[+3[+[5*y[")) | "none")
   write("bal('+','([','])',\"([2*x)+3]+(5*y]\") ----> ",image(bal('+','([','])',"([2*x)+3]+(5*y]")) | "none")
   write("bal(,,,\"()+()\") ----> ",image(bal(,,,"()+()")) | "none")
   write("bal(&cset,,,\"()+()\") ----> ",image(bal(&cset,,,"()+()")) | "none")
   write("bal([],,,\"\") ----> ",image(bal([],,,"")) | "none")
end
