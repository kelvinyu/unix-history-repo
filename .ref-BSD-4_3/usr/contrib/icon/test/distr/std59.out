any('aeiou',&lcase) ----> 2
any('aeiou',&ucase) ----> none
2
2
2
2
2
2
2
2
2
2
2
6
6
6
6
6
6
6
6
6
6
6
every write(any('aeiou',&lcase,1 to 5,10 to 20)) ----> none
match("abc","abcabcabcabc") ----> 4
match("abc","xabcabcabcabc") ----> none
5
5
5
5
5
5
5
8
8
8
5
5
5
8
5
8
5
8
every write(match("abc","xabcabcabcabc",1 to 10,1 to 10)) ----> none
upto('56d&',&lcase) ----> 4
upto('56d&',&ucase) ----> none
upto('56d&',&lcase,15) ----> none
many(&lcase,"this is a Test") ----> 5
many(&lcase,"this is a Test",5) ----> none
many(&lcase,"this is a Test",5,9) ----> none
find("aa","xxaaaaaa") ----> 3
3
4
5
6
7
every write(find("aa","xxaaaaaa")) ----> none
4
5
every write(find("aa","xxaaaaaa",4,7)) ----> none
bal('-','(',')',"-35") ----> 1
bal('+','(',')',"((2*x)+3)+(5*y)") ----> 10
10
7
7
10
every write(bal('+','(',')',"((2*x)+3)+(5*y)",1 to 10)) ----> none
bal('+','[','[',"[[2*x[+3[+[5*y[") ----> none
bal('+','([','])',"([2*x)+3]+(5*y]") ----> 10
bal(,,,"()+()") ----> 1
bal(&cset,,,"()+()") ----> 1
Run-time error 104 at line 31 in std59.icn
cset expected
offending value: []
